### DISCUSSION TITLE
Navigation in Alarm Area Panel

### DISCUSSION THREAD

**Discussion by georgweiss** [ControlSystemStudio/phoebus#2917] [Ideas]
Requested from user: double-click on an item in the Alarm Area Panel view should replace that item with the status of its child items (though not PV nodes). Some UI element would be needed to navigate back.

Of course this in part mimics the functionality of the Alarm Tree, to which the Alarm Area Panel context menu links. Currently though there is a bug: the context menu always launches the Alarm Tree for the default config (but I will fix that).

**Comment by georgweiss** [comment]
@kasemir, @shroffk, @jembishop, @tynanford, @katysaintin...

Any input here?

**Comment by kasemir** [comment]
The purpose of the alarm area panel is that you can see from across the room if everything is OK.
There's a preference setting to select the level that it shows, with the default of  `alarm_area_level=2` listing the top-level "area" elements just below root. I'd say it's important to keep it that way, so operators can be assured to always look at the same thing.
Turning the alarm area panel into a half-baked alarm tree where you can drill down the hierarchy means that user A will click around, then leave, and user B now sees "all green" across the room, unaware that they're only looking at a subsection of the alarm tree.
You may be able to alleviate that by somehow making it very obvious that the area panel no longer shows the default level, but in the end you'll duplicate the alarm tree.
If your user wants a new thing that's somewhere between the area panel and alarm tree, feel free to implement a site-specific new type of alarm display.

**Comment by georgweiss** [comment]
Thanks @kasemir for your input.

**Comment by kasemir** [comment]
> Currently though there is a bug: the context menu always launches the Alarm Tree for the default config (but I will fix that).

Thanks for finding that, and yes, would be good to fix it.

**Comment by jembishop** [comment]
This sort of view is similar to an interactive [treemap](https://towardsdatascience.com/make-a-treemap-in-python-426cee6ee9b8). I don't know much about JavaFX but doesn't seem to have this widget type built in and it would probably make sense to build that first if we wanted this. One example here https://plotly.com/python/treemaps/ implements backwards navigation by displaying a representation of the current path at the top.

**Comment by georgweiss** [comment]
While the tree map look nice it would in my view not add any new feature compared to the Alarm Tree. Except that all non-leaf nodes would be expanded by default.

**Comment by tynanford** [comment]
Our use case for the alarm area panel is placing it on a large TV the operators can view across the control room so no one really interacts with the application besides opening it for the first time. So I'm not opposed but don't see the use case for our site
